animation_nodes/nodes/mesh/get_vertex_color_layer.py

```diff
@@ -0,0 +1,60 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode
+from ... data_structures import Color, ColorList, VirtualColorList
+from .. color.c_utils import getVertexColorsFromLoopColors, getPolygonColorsFromLoopColors
+
+colorModeItems = [
+    ("LOOP", "Loop", "Get color of every loop vertex", "NONE", 0),
+    ("VERTEX", "Vertex", "Get color of every vertex", "NONE", 1),
+    ("POLYGON", "Polygon", "Get color of every polygon", "NONE", 2)    
+]
+
+class GetVertexColorLayerNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GetVertexColorLayerNode"
+    bl_label = "Get Vertex Color Layer"
+    bl_width_default = 155
+    errorHandlingType = "EXCEPTION"
+
+    colorMode: EnumProperty(name = "Color Mode", default = "LOOP",
+        items = colorModeItems, update = propertyChanged)
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh", dataIsModified = True)
+        self.newInput("Text", "Name", "colorLayerName")
+
+        self.newOutput("Color List", "Colors", "colors")        
+
+    def draw(self, layout):
+        layout.prop(self, "colorMode", text = "")
+
+    def execute(self, mesh, colorLayerName):
+        if mesh is None: return ColorList()
```

Comment:
We can use an exception here as well.

Reply:
Omar, For input mesh,
https://github.com/JacquesLucke/animation_nodes/blob/419e1e73431b17cb5cfa6cce76acc588321c5092/animation_nodes/nodes/mesh/get_vertex_color_layer.py#L24
due to this, I think mesh can't none, is it not?

Reply:
An input mesh can be None. But if it is None, I would consider this an issue with the node the produced the None value. So maybe you are right, the None check is not needed here.